dask/dataframe/tests/test_io.py

```diff
@@ -1107,6 +1107,24 @@ def test_read_hdf():
         assert (sorted(dd.read_hdf(fn, '/data').dask) ==
                 sorted(dd.read_hdf(fn, '/data').dask))
 
+def test_read_hdf_multiple():
+    pytest.importorskip('tables')
+    df = pd.DataFrame({'x': ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p'],
+                       'y': [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16]},
+                            index=[1., 2., 3., 4., 5., 6., 7., 8., 9., 10., 11., 12., 13., 14., 15., 16.])
+    a = dd.from_pandas(df, 16)
+
+    with tmpfile('h5') as fn:
+        a.to_hdf(fn, '/data*')
+        with pytest.raises(ValueError) as e:
+            r = dd.read_hdf(fn, '/data*', assume_partitioned=True)
+        assert 'entirety' in str(e)
```

Comment:
Maybe we set `chunksise=no_default` instead and, on `dd.read_hdf(foo.hdf', '/x-*')` we set `chunksize=None` and assume that all of the groups fit into ram unless told otherwise but still set `chunksize=1000000` if given only a single file?